### PR TITLE
add a self-contained cpp test

### PR DESCRIPTION
**Pull request by ziyu-guo**
To cover #61

### CODE REVIEW

File: tests/cpp/test_4in2out.cc [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+std::vector<std::vector<float>> RunInference(DLRModelHandle model);
+
+// Based on hcho3's test harness, this executable 
+// load and test a simple TVM model that sums 2 pairs of tensors 
+// and outputs 2 result tensors.
+//    data1 data2     data3 data4
+//       \  /             \  /
+//      output1          output2
+
+int main(int argc, char** argv) {
```
Comment (zhiics) [inline tests/cpp/test_4in2out.cc]: I thin this is not a good way for unit test. We shouldn't expect developers to explicitly provide anything. Let's use gtest directly for different testing cases.

You can find some examples from here:
https://github.com/dmlc/tvm/tree/master/tests/cpp

File: tests/cpp/gen_4in2out.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import sys
+
+import nnvm.symbol as symbol
+import nnvm.graph as graph
+import nnvm.compiler
+from nnvm.frontend import from_mxnet
+
+import tvm
+
+data1 = symbol.Variable(name="data1")
```
Comment (zhiics) [inline tests/cpp/gen_4in2out.py]: Why do we need this file? It looks to me that they just generate the artifact. We probably should generate them only when we are doing the tests.

File: CMakeLists.txt [hunk 3]
```diff
@@ -203,22 +207,25 @@ add_custom_target(combined_lib
 
 # Tests
 set(TEST_EXECS "")
-file(GLOB TEST_SRCS test/cpp/*.cc)
+file(GLOB TEST_SRCS tests/cpp/*.cc)
 find_library(GTEST_LIB NAMES libgtest.a gtest)
 
-if(GTEST_LIB)
```
Comment (zhiics) [inline CMakeLists.txt]: Why do we remove gtest? I think we need to keep them. And I think we need to tests for DLR using gtests.

**Comment by zhiics** [comment]
Does #75 depend on this?

**Comment by ziyu-guo** [comment]
> Does #75 depend on this?

No. #75 is a follow-up after addressing missing test_assign_op artifacts on arm devices.

File: CMakeLists.txt (hunk 3 above)
Comment (ziyu-guo) [reply CMakeLists.txt]: This c test is supposed to be run on both linux and android, using gtest adds an additional dependency.

**Pull request closed without merging**